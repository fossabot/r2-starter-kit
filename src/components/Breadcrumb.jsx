import React, { PropTypes } from 'react';
import { Link } from 'react-router-dom';

function Breadcrumb({ data }) {
    return (
        <div className="ts breadcrumb">
            {data.map((val, key) => {
                const lastOne = key === data.length - 1;
                return (
                    <div key={`crumb${key}`} className={`${lastOne ? 'active' : ''} section`}>
                        {val.link
                            ? <Link to={val.link}>{val.text}</Link>
                            : val.text
                        }
                        {!lastOne && <i className="right angle icon divider"></i>}
                    </div>
                );
            })}
        </div>
    );
}

Breadcrumb.propTypes = {
    data: PropTypes.arrayOf(PropTypes.shape({
        link: PropTypes.string,
        text: PropTypes.string
    }))
};

export default Breadcrumb;
