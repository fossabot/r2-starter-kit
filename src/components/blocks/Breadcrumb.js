import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

export default class Breadcrumb extends PureComponent {
  static propTypes = {
    title: PropTypes.string,
    handleClick: PropTypes.func,
  };
  static defaultProps = {
    title: '',
    handleClick: () => {},
  };
  render() {
    const { title, handleClick } = this.props;
    return (
      <div className="ts breadcrumb">
        <div className="section">
          <a
            onClick={handleClick}
            style={{ cursor: 'pointer' }}
          >
            Post List
          </a>
        </div>
        <div className="section active">
          <i className="right angle icon divider" />
          {title}
        </div>
      </div>
    );
  }
}
