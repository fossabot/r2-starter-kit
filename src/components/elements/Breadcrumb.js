import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

export default class Breadcrumb extends PureComponent {
  static propTypes = {
    data: PropTypes.arrayOf(
      PropTypes.shape({
        link: PropTypes.string,
        text: PropTypes.string,
      }),
    ),
  };
  static defaultProps = {
    data: [],
  };
  render() {
    const { data } = this.props;
    return (
      <div className="ts breadcrumb">
        {data.map((val, key) => {
          const lastOne = key === data.length - 1;
          return (
            <div
              key={`crumb${key}`}
              className={`section ${lastOne ? 'active' : ''}`}
            >
              {val.link ? <Link to={val.link}>{val.text}</Link> : val.text}
              {!lastOne && <i className="right angle icon divider" />}
            </div>
          );
        })}
      </div>
    );
  }
}
