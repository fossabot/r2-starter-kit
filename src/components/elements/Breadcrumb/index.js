import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

export default class Breadcrumb extends PureComponent {
  static propTypes = {
    data:     PropTypes.arrayOf(PropTypes.shape({
      id:          PropTypes.number,
      text:        PropTypes.string,
      handleClick: PropTypes.func,
    })),
    activeId: PropTypes.number,
  };
  static defaultProps = {
    data:     [],
    activeId: null,
  };
  render() {
    const {
      data,
      activeId,
    } = this.props;
    return (
      <div className="ts breadcrumb">
        {data.map((val, key) => {
          const isActive = val.id === activeId;
          return (
            <div
              key={`breadcrumb${key}`}
              className={cx('section', { 'active': isActive })}
            >
              {key > 0 &&
                <i className="right angle icon divider" />
              }
              {isActive && val.text}
              {!isActive &&
                <a onClick={val.handleClick}>
                  {val.text}
                </a>
              }
            </div>
          );
        })}
      </div>
    );
  }
}
