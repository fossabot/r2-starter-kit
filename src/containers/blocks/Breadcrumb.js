import { connect } from 'react-redux';

import Breadcrumb from 'components/blocks/Breadcrumb';
import { clickRootPath } from 'sagas/panels/post';

function mapStateToProps(state) {
  return state;
}

function mapDispatchToProps(dispatch) {
  return { dispatch };
}

function mergeProps(state, { dispatch }, ownProps) {
  return {
    ...ownProps,

    // custom props
    title: state.response.post.data.title,
    handleClick: () => dispatch(clickRootPath()),
  };
}

export default connect(mapStateToProps, mapDispatchToProps, mergeProps)(Breadcrumb);
