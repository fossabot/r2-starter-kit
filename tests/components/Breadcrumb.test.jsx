import React from 'react';
import { shallow } from 'enzyme';
import Breadcrumb from 'components/Breadcrumb';

describe('<Breadcrumb />', () => {
  it('should render initial component', () => {
    const mockProps = {
      data: [
        { text: 'Post List', link: '/' },
        { text: 'Content' }
      ]
    };
    const wrapper = shallow(
      <Breadcrumb {...mockProps} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
