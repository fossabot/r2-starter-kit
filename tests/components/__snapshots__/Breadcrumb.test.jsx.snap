// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumb /> should render initial component 1`] = `
<div
  className="ts breadcrumb"
>
  <div
    className=" section"
    key="crumb0"
  >
    <Link
      replace={false}
      to="/"
    >
      Post List
    </Link>
    <i
      className="right angle icon divider"
    />
  </div>
  <div
    className="active section"
    key="crumb1"
  >
    Content
  </div>
</div>
`;
